core/dragging/block_drag_strategy.ts

```diff
@@ -120,6 +121,10 @@ export class BlockDragStrategy implements IDragStrategy {
     }
     this.block.setDragging(true);
     this.workspace.getLayerManager()?.moveToDragLayer(this.block);
+
+    // Since moving the block to the drag layer will cause it to lose focus,
+    // ensure it regains focus (to enable the block's selection highlight).
+    getFocusManager().focusNode(this.block);
```

Comment:
Question: should this be done in `moveToDragLayer` (and `moveOffDragLayer`) instead of in the drag strategy?

Reply:
Good idea. This actually made me realize that workspace comments had the same problem, so this shift actually fixes those, too.